app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,15 @@
+class AdminController < ApplicationController
+  before_action :authenticate_admin!
+
+  def orders
+    @orders = Order.all
+  end
+
+  private
+  
+  def authenticate_admin! 
+    unless current_user.role == "admin"
+      redirect_to root_path, alert: "Access denied."
```

Comment:
You can also make this a 1-liner (unless Rubocop complains it's too long):
`redirect_to root_path, alert: "Access denied." unless current_user.role == "admin"`